functions/jwt/workload.py

```diff
@@ -0,0 +1,37 @@
+from syscalls import ResponseStr
+import json
+
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization
+import jwt
+import datetime
+
+PEM_FILE=['home', 'faasten,faasten', 'private_key']
+def handle(syscall, payload=b'', blobs={}, invoker=[], **kwargs):
+    request = json.loads(payload)
+    sub = request['sub']
+    # the invoker should be of length 1 and the tokens list should also be of length 1
+    idp = invoker[0].tokens[0]
```

Comment:
What is `idp` here? I assume it represents the authentication service that is used to auth `sub` (i.e., princeton_cas), correct? Also, what is `invoker`? Is it generated by the gate, and why it is a list?

Reply:
identity provider.
Here, the assumption is that the invoker is gonna have a privilege, looks like "princeton.edu". That is, a CNF with one clause and the clause has only one term.

Reply:
Got it. Is the invoker pre-determined by faasten (aka no need to be specified by the caller)?

Reply:
when the login function get installed it is granted a privilege, which decided the invoker identity.